odoo/addons/base/models/res_partner.py

```diff
@@ -197,7 +197,7 @@ def default_get(self, default_fields):
     title = fields.Many2one('res.partner.title')
     parent_id = fields.Many2one('res.partner', string='Related Company', index=True)
     parent_name = fields.Char(related='parent_id.name', readonly=True, string='Parent name')
-    child_ids = fields.One2many('res.partner', 'parent_id', string='Contact', domain=[('active', '=', True)])
+    child_ids = fields.One2many('res.partner', 'parent_id', string='Contact', domain=[('active', '=', True)], context={'active_test': False})
```

Comment:
I think that it is strange to be forced to add this context as a performance improvement, it should be managed correctly by the orm in this case if possible?
I don't understand why this domain is useful to be honest, isn't it the default behavior ? 
After looking at the history it was added to bypass a _search hack that was removed in 62a1674412d5eeb2ce8d69ef8019a7f8946b99e5, removing also the comment explaining why domain is present so we could maybe just remove the domain here?

Reply:
> I think that it is strange to be forced to add this context as a performance improvement, it should be managed correctly by the orm in this case if possible?
It can be, but it is too niche to do in the ORM, IMO.
> I don't understand why this domain is useful to be honest, isn't it the default behavior ?
No really, by default, the one2many values depends of active_test in the context when you get it from the cache. Here because of the domain (done during the read), `active_test` will never affect the values.

But you might be right that this domain is not relevant anymore (I will try to remove it), but it will affect the current behavior.

Reply:
Removing the domain change behavior of child_ids and breaks one test: test_09_update_cart_address (https://runbot.odoo.com/runbot/build/63167548). Not sure if it is relevant or not but I rather prefer avoiding changing any behavior here.